packages/cloud/connection_details/kibana/kibana_connection_details_provider.tsx

```diff
@@ -65,7 +68,18 @@ const createOpts = async (props: KibanaConnectionDetailsProviderProps) => {
           },
         };
       },
-      hasPermission: async () => true,
+      hasPermission: security
+        ? async () => {
+            try {
+              return (await security.authz.getCurrentUserApiKeyPrivileges()).canManageOwnApiKeys;
+            } catch (error) {
+              if (error instanceof Error && error.message === 'Forbidden') {
```

Comment:
Nit
```suggestion
              if (error instanceof errors.ResponseError && error.statusCode === 403) {
```

Reply:
`import { errors } from '@elastic/elasticsearch'`

Reply:
Yes, that something we use on the server only. I think checking for `Forbidden` message works fine for my case.